src/overlays/actors/ovl_Bg_Spot15_Rrbox/z_bg_spot15_rrbox.c

```diff
@@ -21,45 +34,340 @@ const ActorInit Bg_Spot15_Rrbox_InitVars = {
+    sp2C.y = D_808B45DC[arg2].y * (this->dyna.actor.scale.y * 10.0f);
+    sp2C.z = D_808B45DC[arg2].z * (this->dyna.actor.scale.z * 10.0f);
+
+    func_808B39E8(&sp38, &sp2C, this->unk_16C, this->unk_170);
+
+    sp38.x += this->dyna.actor.posRot.pos.x;
+    sp38.y += this->dyna.actor.pos4.y;
+    sp38.z += this->dyna.actor.posRot.pos.z;
+
+    this->dyna.actor.groundY =
+        func_8003CA64(&globalCtx->colCtx, &this->dyna.actor.floorPoly, &this->bgId, &this->dyna.actor, &sp38, chkDist);
+
+    if (-0.001f <= (this->dyna.actor.groundY - this->dyna.actor.posRot.pos.y)) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.groundY;
+        return true;
+    }
+    return false;
+}
+
+f32 func_808B3DDC(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    int i;
+    Vec3f tempVector2;
+    Vec3f tempVector1;
+    Actor* actor = &this->dyna.actor;
+    f32 yIntersect;
+    f32 returnValue = -32000.0f;
+    s32 bgId;
+
+    func_808B3A34(this);
+    for (i = 0; i < ARRAY_COUNT(D_808B45DC); i++) {
+        tempVector1.x = D_808B45DC[i].x * (actor->scale.x * 10.0f);
+        tempVector1.y = D_808B45DC[i].y * (actor->scale.y * 10.0f);
+        tempVector1.z = D_808B45DC[i].z * (actor->scale.z * 10.0f);
+
+        func_808B39E8(&tempVector2, &tempVector1, this->unk_16C, this->unk_170);
+
+        tempVector2.x += actor->posRot.pos.x;
+        tempVector2.y += actor->pos4.y;
+        tempVector2.z += actor->posRot.pos.z;
+
+        yIntersect = func_8003CA64(&globalCtx->colCtx, &actor->floorPoly, &bgId, actor, &tempVector2, 0);
+
+        if (returnValue < yIntersect) {
+            returnValue = yIntersect;
+            this->bgId = bgId;
+        }
+    }
+    return returnValue;
+}
+
+s32 func_808B3F58(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    if (func_808B3CA0(this, globalCtx, 0)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 1)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 2)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 3)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 4)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_808B4010(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    return func_800435D8(globalCtx, &this->dyna, this->dyna.actor.scale.x * 290.0f,
+                         this->dyna.actor.scale.x * 290.0f + 20.0f, 1.0f) == 0;
+}
+
+void func_808B4084(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_808B40AC;
+    this->dyna.actor.gravity = 0.0f;
+    this->dyna.actor.velocity.x = 0.0f;
+    this->dyna.actor.velocity.y = 0.0f;
+    this->dyna.actor.velocity.z = 0.0f;
+}
+
+void func_808B40AC(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->unk_168 <= 0 && fabsf(this->dyna.unk_150) > 0.001f) {
+        if (func_808B3AAC(this, globalCtx) != 0 && func_808B4010(this, globalCtx) == 0) {
```

Comment:
`func_808B4010` 's result can be treated as a bool as well (applies later in the file too)